codalab/server/bundle_manager.py

```diff
@@ -118,6 +118,13 @@ def _stage_bundles(self):
         for bundle in bundles:
             parent_uuids = set(dep.parent_uuid for dep in bundle.dependencies)
 
+            missing_uuids = parent_uuids - all_parent_uuids
```

Comment:
There was a bug in the BundleManager staging logic in which if parent bundles are missing, an "insufficient permissions" error comes up instead of a "missing parent bundles" error.

I came across this when writing one of the tests for this function, and that test now checks for the right error message.